File: benchmark/c/cpw/fibonacci_modulo_p/fibonacci_modulo_p.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int fibonacci_modulo_p ( int p ) ;
int fibonacci_modulo_p ( int p ) {
  int first = 1, second = 1, number = 2, next = 1;
  while ( next ) {
    next = ( first + second ) % p;
    first = second;
    second = next;
    number ++;
  }
  return number;
}

File: translation_result/else_bms/fail/c/cpw/fibonacci_modulo_p/gemini/Restart/fibonacci_modulo_p.rs

fn fibonacci_modulo_p(p: i32) -> i32 {
    let mut first = 1;
    let mut second = 1;
    let mut number = 2;
    let mut next = 1;

    while next != 0 {
        next = (first + second) % p;
        first = second;
        second = next;
        number += 1;
    }
    number
}
